src/swarm-lang/Swarm/Language/Typecheck.hs

```diff
@@ -518,7 +518,7 @@ instance PrettyPrec TypeErr where
     EscapedSkolem x ->
       "Skolem variable" <+> ppr x <+> "would escape its scope"
     UnboundVar x ->
-      "Unbound variable" <+> ppr x
+      "Undefined variable" <+> ppr x
```

Comment:
Out of curiosity, is there a difference between the two terms?

I guess "undefined" is more familiar and you did not rename `UnboundVar` to not have a large diff?

Reply:
There is not really much a difference.  "Unbound" is more technically precise, because e.g. a variable could be defined *somewhere* but not bound in the specific scope where the error happened.  I kept `UnboundVar` because it is a precise description of the error that happened; I just changed the way the error is displayed to use a word that is likely to be more familiar to players who are not programming language nerds.